tutorials/klusterai-api/effective-prompt-engineering.ipynb

```diff
@@ -0,0 +1,731 @@
+    "- Clearly specify which products to prioritize in each channel and explain why.\n",
+    "\n",
+    "**Structure your response clearly:**\n",
+    "- Use separate headings for each of the above sections.\n",
+    "- Present recommendations clearly in bullet-point format.\n",
+    "- Briefly but explicitly explain your reasoning behind each suggestion.\n",
+    "\n",
+    "Please be concise, specific, and actionable.\"\"\"\n",
+    "\n",
+    "# Display the prompt for visual reference\n",
+    "display(Markdown(best_prompt_text))\n",
+    "\n",
+    "# Use the variable in the API call\n",
+    "response_best = client.chat.completions.create(\n",
+    "    model=model,\n",
+    "    messages=[{\"role\": \"user\", \"content\": best_prompt_text}],\n",
+    "    temperature=0.5\n",
+    ").choices[0].message.content\n",
+    "print(\"\\nModel response:\\n\" + response_best)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3455c366",
+   "metadata": {},
+   "source": [
+    "### Analysis\n",
+    "The Best prompt excels because it integrates comprehensive data (product sales, prices, and profit margins) with detailed business context (target demographics, product characteristics, marketing channels, and specific challenges). By explicitly specifying the desired structure—including step-by-step revenue and profit analysis, targeted profitability recommendations, and marketing optimizations—it guides the LLM toward high-value, actionable insights.\n",
+    "\n",
+    "Key strengths of this Best prompt:\n",
+    "1. **Complete Data & Metrics**\n",
+    "   * Monthly sales, pricing, and profit margins remove guesswork and allow precise revenue/profit calculations.\n",
+    "2. **Focused Business Context**\n",
+    "   * Target customer demographics, marketing channels, and product nuances (e.g., overstock of Product C) align the output with real business priorities.\n",
+    "3. **Clear Task Breakdown**\n",
+    "   * Dividing the request into distinct sections (Revenue/Profit Analysis, Profitability Recommendations, Marketing Strategies) ensures the response is both methodical and thorough.\n",
+    "4. **Actionability & Specificity**\n",
+    "   * The prompt requests concrete next steps—such as pricing tweaks and promotional ideas—tailored to each product's characteristics and customer segment.\n",
+    "5. **Logical Reasoning Instructions**\n",
+    "   * Guiding the LLM to \"think step-by-step\" fosters more thoughtful planning, especially around clearing excess inventory and leveraging different marketing channels.\n",
+    "\n",
+    "Overall, this approach maximizes the LLM's potential, yielding responses that are precise, context-aware, and immediately useful to decision-makers."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "22c05fc0-6f1c-4137-8d5b-fe36c8401c5c",
+   "metadata": {},
+   "source": [
+    "## Leveraging AI to improve your prompts\n",
```

Comment:
I thought this section would be to show how can we actually use AI to generate the prompt for the example above. Once again, reiterate the goal to the AI and the data you have available and see if it comes with a good prompt, and compare it with your previous :)

Reply:
Revised !